src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/util/OcrDataSerializationJourneyTest.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.util;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.IOUtils;
+import org.junit.Test;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.Envelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.helper.EnvelopeCreator;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.blob.InputEnvelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.mapper.EnvelopeMapper;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.out.msg.EnvelopeMsg;
+
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+public class OcrDataSerializationJourneyTest {
+
+    @Test
+    public void should_deserialize_all_ocr_fields_in_insertion_order() throws Exception {
```

Comment:
I think this test should be an integration test, as it checks the whole journey. It should also include saving and reading from the DB, as that's a very uncertain bit here.

Reply:
Moved the test to integration tests and added assertions for the envelope read from db